common/src/com/marvelousbob/common/network/register/dto/EnemyType.java

```diff
@@ -1,8 +1,33 @@
 package com.marvelousbob.common.network.register.dto;
 
+import com.badlogic.gdx.graphics.Color;
+import com.badlogic.gdx.math.Vector2;
+import com.marvelousbob.common.model.entities.dynamic.enemies.Enemy;
+import com.marvelousbob.common.model.entities.dynamic.enemies.PolygonEnemy;
+import com.marvelousbob.common.utils.UUID;
+
 public enum EnemyType {
-    TRIANGLE,
-    SPIKED_CIRCLE,
-    MISSILE,
-    RANGED
-}
+    POLYGON {
+        public Enemy getPlayerInstance(UUID uuid, UUID spawnPoint, Vector2 initPos, Color color) {
+            return new PolygonEnemy(uuid, spawnPoint, initPos, color);
+        }
+    }/*,
+    SPIKED_CIRCLE {
+        public Enemy getPlayerInstance(UUID uuid, Color color, Vector2 initPos) {
+            return new RangedPlayer(uuid, color, initPos);
+        }
+    },
+    MISSILE {
+        public Enemy getPlayerInstance(UUID uuid, Color color, Vector2 initPos) {
+            return new RangedPlayer(uuid, color, initPos);
+        }
+    },
+    RANGED {
+        public Enemy getPlayerInstance(UUID uuid, Color color, Vector2 initPos) {
+            return new RangedPlayer(uuid, color, initPos);
+        }
+    }*/;
+
+    public abstract Enemy getPlayerInstance(UUID uuid, UUID spawnPoint, Vector2 initPos,
```

Comment:
rename to getEnemyInstance

Reply:
Good catch. Mais j'aime vrmt pas trop cette solution de EnemyType. Ça allait bien pour les Players, mais si jamais tu te lances sur le Dto, j'aimerais bien voir ce que tu finis par faire.